src/App.tsx

```diff
@@ -1,8 +1,115 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+import { getUsers, getTodos, Todo } from './helpers/api';
+
+const App = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleLoadClick = async () => {
+    setIsLoading(true);
+
+    const todosFromServer = await getTodos();
+    const usersFromServer = await getUsers();
+
+    const todosWithUsers = todosFromServer.map(todo => ({
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId),
+    }));
+
+    setInitialTodos(todosWithUsers);
+    setTodos(todosWithUsers);
```

Comment:
store only the initial array in the state, the other one should be calculated just before rendering